decompiler/frontend/binaryninja/handlers/assignments.py

```diff
@@ -0,0 +1,176 @@
+        contraction_type = destination_operand.type.resize(assignment.size * self.BYTE_SIZE)
+        return UnaryOperation(OperationType.cast, [destination_operand], vartype=contraction_type, contraction=True)
+
+    def _lift_masked_assignment(self, assignment: mediumlevelil.MediumLevelILSetVarField, **kwargs) -> BinaryOperation:
+        return BinaryOperation(
+            OperationType.bitwise_or,
+            [
+                BinaryOperation(
+                    OperationType.bitwise_and,
+                    [
+                        self._lifter.lift(assignment.prev, parent=assignment),
+                        Constant(
+                            self._get_all_ones_mask_for_type(assignment.dest.var.type.width)
+                            - self._get_all_ones_mask_for_type(assignment.size)
+                            << (assignment.offset * self.BYTE_SIZE)
+                        ),
+                    ],
+                    vartype=self._lifter.lift(assignment.src.expr_type, parent=assignment),
+                ),
+                BinaryOperation(
+                    OperationType.left_shift,
+                    [self._lifter.lift(assignment.src, parent=assignment), Constant(assignment.offset * self.BYTE_SIZE)],
+                    vartype=self._lifter.lift(assignment.src.expr_type, parent=assignment),
+                ),
+            ],
+            vartype=self._lifter.lift(assignment.expr_type, parent=assignment),
+        )
+
+    def _lift_mask_high(self, instruction: mediumlevelil.MediumLevelILSetVarField, **kwargs) -> BinaryOperation:
+        """
+        Lift an instruction masking the higher part of a value.
+        e.g. eax.al = eax & 0x000000ff
+        """
+        return BinaryOperation(
+            OperationType.bitwise_and,
+            [op := self._lifter.lift(instruction.src, parent=instruction), Constant(self._get_all_ones_mask_for_type(instruction.size))],
+            vartype=op.type.resize(instruction.size * self.BYTE_SIZE),
+        )
+
+    def _get_all_ones_mask_for_type(self, type_size: int, **kwargs) -> int:
+        """Generate a bit mask for the given type_size."""
+        return int(2 ** (type_size * self.BYTE_SIZE) - 1)
+
+    def lift_split_assignment(self, assignment: mediumlevelil.MediumLevelILSetVarSplit, **kwargs) -> Assignment:
+        """Lift an instruction writing to a register pair."""
+        return Assignment(
+            RegisterPair(
+                high := self._lifter.lift(assignment.high, parent=assignment),
+                low := self._lifter.lift(assignment.low, parent=assignment),
+                vartype=high.type.resize((high.type.size + low.type.size)),
+            ),
+            self._lifter.lift(assignment.src, parent=assignment),
+        )
+
+    def _lift_store_struct(self, instruction: mediumlevelil.MediumLevelILStoreStruct, **kwargs) -> Assignment:
+        """Lift a MLIL_STORE_STRUCT_SSA instruction."""
```

Comment:
If you have an example out-of-the box, would you mind adding it to the docstring?

Reply:
added in 6fdc837c